src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -1098,7 +1140,18 @@ export class SavedObjectsRepository {
       },
       { ignore: [404] }
     );
-
+    if (
+      isNotFoundFromUnsupportedServer({
+        statusCode: aliasResponse.statusCode,
+        headers: { ...aliasResponse.headers },
```

Comment:
As we aren't mutating headers anywhere in `isNotFoundFromUnsupportedServer`, do we need to spread them here?

(Same question applies to the other places in this file where we are spreading the headers)

Reply:
done (e.g. https://github.com/elastic/kibana/pull/108749/files#diff-01c073903639e29be8562ed0de79fae4084a953304672d1e6bdb3fd7fef06f11R431)